kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/drivers/assets/AssetModel.java

```diff
@@ -40,16 +41,21 @@ public interface AssetModel {
 
     public boolean isValid();
 
+    public Integer getParameterIndex(String value);
+
+    public Map<String, Integer> getParameterIndexes();
+
     public interface ChannelModel {
 
         public GwtConfigParameter getParameter(int index);
 
-        public void setValue(String id, String value);
+        public void setValue(Integer index, String value);
 
-        public boolean isValid(String id);
+        public boolean isValid(Integer index);
 
-        public String getValue(String id);
+        public String getValue(Integer index);
```

Comment:
Why you changed the way the resource is got? Key vs. index?

Reply:
It was required to extract the LegacyChannelModel if I'm not wrong. The class was a mess